public/pages/ThreatIntel/containers/AddThreatIntelSource/AddThreatIntelSource.tsx

```diff
@@ -99,50 +158,155 @@ export const AddThreatIntelSource: React.FC<AddThreatIntelSourceProps> = ({
+    const name = event.target.value;
+    const nameError = validateSourceName(name);
+    setFieldError({
+      name: nameError || '',
+    });
+
     setSource({
       ...source,
-      name: event.target.value,
+      name,
     });
+    setFieldTouched({ name: true });
   };
 
+  const validateSourceDescription = (description: string) => {
+    let error;
+    const isValid = validateDescription(description, THREAT_INTEL_SOURCE_DESCRIPTION_REGEX);
+    if (!isValid) {
+      error = 'Invalid name.';
+    }
+
+    return error;
+  };
   const onDescriptionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const descriptionError = validateSourceDescription(event.target.value);
+    setFieldError({
+      description: descriptionError || '',
+    });
+
     setSource({
       ...source,
       description: event.target.value,
     });
+    setFieldTouched({ description: true });
   };
 
+  const validateS3ConfigField = (value: string) => {
+    if (!value) {
+      return `Required.`;
+    }
+  };
   const onS3DataChange = (field: keyof S3ConnectionSource['s3'], value: string) => {
+    const error = validateS3ConfigField(value);
+    setFieldError({
+      s3: {
+        ...inputErrors.s3,
+        [field]: error || '',
+      },
+    });
     setS3ConnectionDetails({
       s3: {
         ...s3ConnectionDetails.s3,
         [field]: value,
       },
     });
+    setFieldTouched({ s3: { ...inputTouched.s3, [field]: true } });
   };
 
+  const validateSchedule = (schedule: PeriodSchedule) => {
+    setFieldError({
+      schedule:
+        !schedule.period.interval || Number.isNaN(schedule.period.interval)
+          ? 'Invalid schedule'
+          : '',
+    });
+  };
   const onIntervalChange = (schedule: PeriodSchedule) => {
+    validateSchedule(schedule);
     setSchedule({
       interval: {
         start_time: Date.now(),
         period: schedule.period.interval,
         unit: schedule.period.unit,
       },
     });
+    setFieldTouched({ schedule: true });
   };
 
   const onFileChange = (files: FileList | null) => {
-    setFileError('');
+    setFieldError({
+      fileUpload: {
+        ...inputErrors.fileUpload,
+        file: files?.length === 0 ? 'File required.' : '',
+      },
+    });
     if (!!files?.item(0)) {
       readIocsFromFile(files[0], (response) => {
         if (response.ok) {
           setFileUploadSource(response.sourceData);
         } else {
-          setFileError(response.errorMessage);
+          setFieldError({
+            fileUpload: {
+              ...inputErrors.fileUpload,
+              file: response.errorMessage,
+            },
+          });
         }
       });
     }
+    setFieldTouched({ fileUpload: { file: true } });
+  };
+
+  const isThereAnError = (errors: any): boolean => {
+    for (let key of Object.keys(errors)) {
+      if (
+        (sourceType !== 'S3_CUSTOM' && key === 's3') ||
+        (sourceType !== 'IOC_UPLOAD' && key === 'fileUpload') ||
```

Comment:
Not blocking:
Ideally, it would be helpful for these values to be stored in an exhaustive enum somewhere.